src/users/views.py

```diff
@@ -1,56 +1,100 @@
+        context = {
             "user_profile": user,
             "user_author": author,
             "user_models": models,
             "user_predictions": predictions,
             "user_avatar": avatar_url,
-            "form": form,
-        },
-    )
+        }
+        return context
+
+    def get(self, request, username):
+        context = self.get_context_data(username)
+        return render(request, self.template_name, context)
+
+    def post(self, request, username):
+        context = self.get_context_data(username)
+
+        if "update_author" in request.POST:
+            form = UpdateAuthorForm(request.POST)
+            if form.is_valid():
+                user = context["user_profile"]
+                author = context["user_author"]
+                user.first_name = form.cleaned_data["first_name"]
+                user.last_name = form.cleaned_data["last_name"]
+                user.save()
+
+                author.institution = form.cleaned_data["institution"]
+                author.save()
+
+                messages.success(request, "Author updated successfully")
+            else:
+                messages.error(request, "Invalid request")
+
+        elif "update_model" in request.POST:
+            form = UpdateModelForm(request.POST)
+            if form.is_valid():
+                model_id = form.cleaned_data["model_id"]
+
+                payload = {
+                    "model_name": form.cleaned_data["model_name"],
+                    "model_description": form.cleaned_data["model_description"],
+                    "model_repository": form.cleaned_data["model_repository"],
+                    "model_language": form.cleaned_data["model_language"],
+                    "model_type": form.cleaned_data["model_type"],
+                }
+
+                response = self.update_model_api_call(request, model_id, payload)
+
+                if response.status_code == 200:
+                    messages.success(request, "Model updated successfully")
+                elif response.status_code == 401:
+                    messages.error(request, "Unauthorized")
+                else:
+                    print(response.__dict__)
+                    messages.error(
+                        request,
+                        f"Failed to update model: {response.json().get('message')}",
+                    )
+            else:
+                messages.error(request, form.errors.values[0].as_data())
+
+        return redirect("profile", username=username)
+
+    def update_model_api_call(self, request, model_id, payload):
+        api_url = request.build_absolute_uri(
+            reverse("api-1:update_model", kwargs={"model_id": model_id})
+        )
+        # csrf_token = request.COOKIES.get("csrftoken")
+        csrf_token = csrf.get_token(request)
+        headers = {"X-CSRFToken": csrf_token}
+        print(headers)
+        response = requests.put(api_url, json=payload, headers=headers)
```

Comment:
future note: passing csrftoken in the header is not working

Reply:
> auth=django_auth,

This line is responsible for checking for user authentication, supposedly by using the CSRF token, in the "UIDKey" calls (using uuid4 authentication that I made), the requirement to make the POST calls is to pass the user key in the headers. In the base django authentication method, it shouldn't even require the CSRF token explicit, since it's already in the request, but it is not working at all.

Other solution would to this problem would be JWT authentication or not using api calls at all, restricting the Update calls to the Django ORM, since it will only be called withing the author page after logging in. The problem with the last solution is that there are some advantages in using API calls in the entire website. @fccoelho